user_commands/UserCommands.cs

```diff
@@ -0,0 +1,113 @@
+            * \param[in] deviceType The type of device being added
+            * \param[in] houseName The name of the house the decive being added to
+            * \param[in] roomName The name of the room the device is in
+            * \param[in] client Client to send the device information to
+            * \param[out] Flag indicating success
+            */
+            static public bool sendAddDevice(string deviceType, string houseName, string roomName, HttpClient client)
+            {
+                return true;
+            }
+
+            /**
+            * Sends a request to update a preference in the application
+            * \param[in] preferenceName The name of the preference the update is being applied to
+            * \param[in] preferenceSetting The new setting for the preference being updated
+            * \param[in] client Client to send the preference update information to
+            * \param[out] Flag indicating success
+            */
+            static public bool sendUpdatePreference(string preferenceName, string preferenceSetting, HttpClient client)
+            {
+                return true;
+            }
+
+            /**
+            * Sends a request to update a setting of a device
+            * \param[in] deviceID The ID of the device to be updated
+            * \param[in] deviceSetting The setting of the specified device to be changed
+            * \param[in] settingValue The new value of the setting specified 
+            * \param[in] client Client to send the preference update information to
+            * \param[out] Flag indicating success
+            */
+            static public bool sendUpdateDevice(string deviceID, string deviceSetting, string settingValue, HttpClient client)
+            {
+                return true;
+            }
+
+            /**
+            * Checks a setting on a device and gets it's value
+            * \param[in] deviceID The ID of the device to be checked
+            * \param[in] deviceSetting The setting of the specified device to be checked
+            * \param[in] client Client to send the query information to
+            * \param[out] result String containing the value of the device setting checked
+            */
+            static public string sendCheckDevice(string deviceID, string deviceSetting, HttpClient client)
+            {
+                string result = "";
+                return result;
+            }
+
+            /**
+            * Updates a user's current location
+            * \param[in] user The user whose location is being updated
+            * \param[in] newLocation The setting of the specified device to be checked
+            * \param[in] client Client to send the location information to
+            * \param[out] Flag indicating success
+            */
+            static public bool sendUpdateLocation(string user, string newLocation, HttpClient client)
```

Comment:
All of these functions are stringly-typed, which isn't considered good practice. This should be worth a first cut stub however.